packages/suite/src/actions/wallet/tokenActions.ts

```diff
@@ -3,19 +3,36 @@ import { Account } from 'src/types/wallet';
 import { Dispatch } from 'src/types/suite';
 import * as accountUtils from '@suite-common/wallet-utils';
 import { notificationsActions } from '@suite-common/toast-notifications';
-import { accountsActions } from '@suite-common/wallet-core';
+import { accountsActions, updateFiatRatesThunk } from '@suite-common/wallet-core';
+import { Timestamp, TokenAddress } from '@suite-common/wallet-types';
+import { FiatCurrencyCode } from '@suite-common/suite-config';
 
-export const addToken = (account: Account, tokenInfo: TokenInfo[]) => (dispatch: Dispatch) => {
-    dispatch(
-        accountsActions.updateAccount({
-            ...account,
-            tokens: (account.tokens || []).concat(accountUtils.enhanceTokens(tokenInfo)),
-        }),
-    );
+export const addToken =
```

Comment:
By updating account, new token definition should be fetched.
I am thinking if is is necessary to add `isCustomToken` attribute here. (if yes, I would use something like `forceFetchToken`)

I know that the token definition will not be ready in the time when `updateFiatRatesThunk` is called. 
However, we should maybe somehow guarantee that fiat rates for tokens are not called earlier than token definitions are fetched.
3rd bullet https://github.com/trezor/trezor-suite/pull/11120
What do you think?

Reply:
Was not trying it before you PR was merged, already rebased on develop so will try again

Reply:
It really needs to be there if we don't want to do the 3rd bullet that you mentioned. Imho other than for these use case it will be always fetched before so let's not care about it to much.

But better naming is good point, changed it to what you are suggesting - d022ac2aca7704564e38be1cbfd53247d587bff8